v5/patch.go

```diff
@@ -51,7 +55,15 @@ type Operation map[string]*json.RawMessage
 // Patch is an ordered collection of Operations.
 type Patch []Operation
 
-type partialDoc map[string]*lazyNode
+type partialDoc struct {
+	items []docItem
```

Comment:
It occurs to me that I don't actually need the `docItem` type, as all that is needed is the order of the keys - the values are already present in the `obj` map. I'm happy to update this.

Reply:
https://github.com/golang/go/issues/27179 discusses how they might sort this out in `encoding/json`